core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -657,15 +657,19 @@ class KafkaServer(
   }
 
   private def createCurrentControllerIdMetric(): Unit = {
-    KafkaYammerMetrics.defaultRegistry().newGauge(MetadataLoaderMetrics.CURRENT_CONTROLLER_ID, () => {
-      Option(metadataCache) match {
-        case None => -1
-        case Some(cache) => cache.getControllerId match {
```

Comment:
Will this work in hybrid mode? Don't we still need to read from the metadata cache to get the KRaft controller ID once we've entered hybrid mode?

Reply:
It should be OK in hybrid mode since we are still updating the znode in hybrid mode. And this comes from there directly.

Reply:
Ah right, makes sense 👍